PSAlphaFS.psm1

```diff
@@ -525,20 +525,44 @@ function Copy-LongItem
 		}
 			
 		$PathObject = New-Object -TypeName Alphaleonis.Win32.Filesystem.FileInfo -ArgumentList $Path, $PathFSFormatObject::FullPath
+
+        if ($Destination -ne $null) {
+          $Destination.TrimEnd()
+        }
         
 		$basename = $PathFSObject::GetFileName($Path)
 		$dBasename = $PathFSObject::GetFileName($Destination)
 		$dParent = $PathFSObject::GetDirectoryName($Destination)
         
-		$dBasenameObj = New-Object -TypeName Alphaleonis.Win32.Filesystem.FileInfo -ArgumentList $dBasename
-		$isFile = if($PathFSObject::HasExtension($dBasename) )
+		
+		$isFile = if($PathFSObject::HasExtension($dBasename))
 		{
+            #has an extention so destination is a file
+            #Question: how does it handle directory with a period in the name?
 			$true
 		}
 		else 
 		{
-			$false
+            if ($Destination.EndsWith('\')) {
+              #ends with backslash then so destination is a directory
+              $false
```

Comment:
please add the following because the methods wont work with a leading '\'
                $Destination = $Destination.TrimEnd('\')
                $dBasename = $PathFSObject::GetFileName($Destination)
                $dParent = $PathFSObject::GetDirectoryName($Destination)